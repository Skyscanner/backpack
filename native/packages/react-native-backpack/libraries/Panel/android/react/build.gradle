apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

android {
  compileSdkVersion rootProject.ext.compileSdkVersion
  buildToolsVersion rootProject.ext.buildToolsVersion


  defaultConfig {
    minSdkVersion rootProject.ext.minSdkVersion
    targetSdkVersion rootProject.ext.targetSdkVersion
    versionCode 1
    versionName "1.0"
  }
}

dependencies{
  implementation "com.facebook.react:react-native:+" // From node_modules
  implementation project(':panel-native')
  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
}
repositories {
  mavenCentral()
}
