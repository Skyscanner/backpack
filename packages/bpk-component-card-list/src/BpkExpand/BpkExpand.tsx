/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { BUTTON_TYPES, BpkButtonV2 } from '../../../bpk-component-button';
import {
  withButtonAlignment,
  withRtlSupport,
} from '../../../bpk-component-icon';
import ChevronDown from '../../../bpk-component-icon/sm/chevron-down';
import ChevronUp from '../../../bpk-component-icon/sm/chevron-up';

import type { ExpandProps } from '../common-types';

const AlignedChevronDownIcon = withButtonAlignment(withRtlSupport(ChevronDown));
const AlignedChevronUpIcon = withButtonAlignment(withRtlSupport(ChevronUp));

const BpkExpand = ({
  children,
  collapsed,
  hideContent,
  setCollapsed,
  showContent,
}: ExpandProps) => {
  const buttonIcon = collapsed ? (
    <AlignedChevronDownIcon />
  ) : (
    <AlignedChevronUpIcon />
  );
  const buttonOnClick = () => {
    if (collapsed) {
      showContent();
      setCollapsed(false);
    } else {
      hideContent();
      setCollapsed(true);
    }
  };

  return (
    <BpkButtonV2
      data-testid="expand-button"
      type={BUTTON_TYPES.link}
      onClick={() => buttonOnClick()}
      aria-expanded={!collapsed}
    >
      {children}
      {buttonIcon}
    </BpkButtonV2>
  );
};

export default BpkExpand;
