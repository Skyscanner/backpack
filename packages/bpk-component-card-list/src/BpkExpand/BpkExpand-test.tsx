/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import BpkExpand from './BpkExpand';

describe('BpkExpand', () => {
  const hideContent = jest.fn();
  const setCollapsed = jest.fn();
  const showContent = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly when collapsed', async () => {
    const user = userEvent.setup();

    render(
      <BpkExpand
        collapsed
        hideContent={hideContent}
        setCollapsed={setCollapsed}
        showContent={showContent}
      >
        Expand
      </BpkExpand>,
    );

    const button = screen.getByTestId('expand-button');
    expect(button).toHaveTextContent('Expand');

    await user.click(button);
    expect(showContent).toHaveBeenCalled();
    expect(setCollapsed).toHaveBeenCalledWith(false);
    expect(hideContent).not.toHaveBeenCalled();
  });

  it('should render correctly when expanded', async () => {
    const user = userEvent.setup();

    render(
      <BpkExpand
        collapsed={false}
        hideContent={hideContent}
        setCollapsed={setCollapsed}
        showContent={showContent}
      >
        Collapse
      </BpkExpand>,
    );

    const button = screen.getByTestId('expand-button');
    expect(button).toHaveTextContent('Collapse');

    await user.click(button);
    expect(hideContent).toHaveBeenCalled();
    expect(setCollapsed).toHaveBeenCalledWith(true);
    expect(showContent).not.toHaveBeenCalled();
  });
});
