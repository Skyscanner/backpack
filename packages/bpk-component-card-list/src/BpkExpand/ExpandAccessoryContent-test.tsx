/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import ExpandAccessoryContent from './ExpandAccessoryContent';

describe('ExpandAccessoryContent', () => {
  const onExpandToggle = jest.fn();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly when collapsed', async () => {
    const user = userEvent.setup();

    render(
      <ExpandAccessoryContent collapsed onExpandToggle={onExpandToggle}>
        Expand
      </ExpandAccessoryContent>,
    );

    const button = screen.getByTestId('bpk-card-list__accessory-expand-button');
    expect(button).toHaveTextContent('Expand');

    await user.click(button);
    expect(onExpandToggle).toHaveBeenCalled();
  });

  it('should render correctly when expanded', async () => {
    const user = userEvent.setup();

    render(
      <ExpandAccessoryContent collapsed={false} onExpandToggle={onExpandToggle}>
        Collapse
      </ExpandAccessoryContent>,
    );

    const button = screen.getByTestId('bpk-card-list__accessory-expand-button');
    expect(button).toHaveTextContent('Collapse');

    await user.click(button);
    expect(onExpandToggle).toHaveBeenCalled();
  });
});
