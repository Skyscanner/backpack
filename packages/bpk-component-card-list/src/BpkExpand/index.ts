import BpkExpand from "./BpkExpand";

export default BpkExpand;
