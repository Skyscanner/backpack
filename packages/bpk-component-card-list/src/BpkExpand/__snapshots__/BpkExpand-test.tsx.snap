// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkExpand should call hideContent function when not collapsed and button is clicked 1`] = `
<DocumentFragment>
  <button
    class="bpk-button bpk-button--link"
    data-testid="button"
    type="button"
  >
    Show More
    <span
      style="line-height: 1rem; display: inline-block; margin-top: 0.25rem; vertical-align: top;"
    >
      <svg
        aria-hidden="true"
        class="bpk-icon--rtl-support"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.887 15.905a1.496 1.496 0 010-2.008l6.397-5.948a1 1 0 011.358-.004l6.532 6.012a1.427 1.427 0 01-.138 1.948 1.572 1.572 0 01-1.997.103L11.96 11.37l-4.97 4.535a1.72 1.72 0 01-2.104 0z"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`BpkExpand should call showContent function when collapsed and button is clicked 1`] = `
<DocumentFragment>
  <button
    class="bpk-button bpk-button--link"
    data-testid="button"
    type="button"
  >
    Show More
    <span
      style="line-height: 1rem; display: inline-block; margin-top: 0.25rem; vertical-align: top;"
    >
      <svg
        aria-hidden="true"
        class="bpk-icon--rtl-support"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.113 8.095a1.496 1.496 0 010 2.008l-6.397 5.948a1 1 0 01-1.358.003l-6.532-6.01a1.427 1.427 0 01.138-1.949 1.572 1.572 0 011.997-.103l5.078 4.638 4.97-4.535a1.72 1.72 0 012.104 0z"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`BpkExpand should render correctly 1`] = `
<DocumentFragment>
  <button
    class="bpk-button bpk-button--link"
    data-testid="button"
    type="button"
  >
    Show More
    <span
      style="line-height: 1rem; display: inline-block; margin-top: 0.25rem; vertical-align: top;"
    >
      <svg
        aria-hidden="true"
        class="bpk-icon--rtl-support"
        style="width: 1rem; height: 1rem;"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.113 8.095a1.496 1.496 0 010 2.008l-6.397 5.948a1 1 0 01-1.358.003l-6.532-6.01a1.427 1.427 0 01.138-1.949 1.572 1.572 0 011.997-.103l5.078 4.638 4.97-4.535a1.72 1.72 0 012.104 0z"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;
